### PR TITLE
Updates for USB host API

### PR DESCRIPTION
**Pull request by lucianomartin**
- add list_connected_devices()
- silence warning for Windows

All the changes are backward compatible and they have been tested on both Windows and MacOS.

### CODE REVIEW

File: lib_device_control/host/device_access_usb.c [hunk 1]
```diff
@@ -33,6 +33,7 @@ static const int sync_timeout_ms = 100;
 void debug_libusb_error(int err_code)
 {
 #if defined _WIN32
+  (void) err_code;
```
Comment (chrisc-xmos) [inline lib_device_control/host/device_access_usb.c]: What does this do?

File: lib_device_control/host/device_access_usb.c [hunk 2]
```diff
@@ -175,6 +176,64 @@ control_read_command(control_resid_t resid, control_cmd_t cmd,
...
+  for (struct usb_bus *bus = usb_get_busses(); bus && !devh; bus = bus->next) {
+    for (struct usb_device *dev = bus->devices; dev; dev = dev->next) {
+      if (dev->descriptor.idVendor == vendor_id) {
+        devh = usb_open(dev);
+        if (!devh) {
+          fprintf(stderr, "Failed to open device with Product ID: %#04x\n", dev->descriptor.idProduct);
+          return CONTROL_ERROR;
+        }
+        current_pid = dev->descriptor.idProduct;
+  #else
```
Comment (chrisc-xmos) [inline lib_device_control/host/device_access_usb.c]: This is just too hard to read. Better to split it into architecture specific functions

File: lib_device_control/host/device_access_usb.c [hunk 3]
```diff
@@ -175,6 +176,64 @@ control_read_command(control_resid_t resid, control_cmd_t cmd,
...
+
+  libusb_device **devs = NULL;
+  int num_dev = libusb_get_device_list(NULL, &devs);
+  struct libusb_device_descriptor desc;
+  for (int dev_idx=0; dev_idx<num_dev; dev_idx++) { 
+    libusb_get_device_descriptor(devs[dev_idx], &desc);
+    if (desc.idVendor == vendor_id) {
+        current_pid = desc.idProduct;
+  #endif        
+        if (num_devices+1>=MAX_NUM_DEVICES) {
```
Comment (chrisc-xmos) [inline lib_device_control/host/device_access_usb.c]: Shouldn't this be > instead of >= ?

File: lib_device_control/host/device_access_usb.c [hunk 4]
```diff
@@ -175,6 +176,64 @@ control_read_command(control_resid_t resid, control_cmd_t cmd,
...
+          fprintf(stderr, "Maximum number of connected devices reached: %d\n", MAX_NUM_DEVICES);
+          return CONTROL_ERROR;
+        }
+        pid_list[num_devices] = current_pid;
+        num_devices++;
```
Comment (chrisc-xmos) [inline lib_device_control/host/device_access_usb.c]: Abstracting lines 212 - 217 in a way that combines them with 228 - 232 might work. The resulting function could be used in both _WIN32 and non _WIN32 implementations and would remove the need for a fixed size buffer.

File: lib_device_control/host/device_access_usb.c (hunk 2 above)
Comment (chrisc-xmos) [reply lib_device_control/host/device_access_usb.c]: Just two separate code segments that are conditionally included would help readability.

**Comment by lucianomartin** [comment]
I need to do some more testing on Windows. I will close this PR and reopen it once the testing is completed.

**Pull request closed without merging**